### PR TITLE
Create PHP tech docs

### PR DESCRIPTION
**Pull request by jonathanglassman**
## What problem does the pull request solve?

Creating PHP tech docs. Requires review and pairing from PHP developer.

## Checklist

- [x] I’ve used the pull request template
- [ ] I’ve written unit tests for these changes
- [x] I’ve updated the documentation (in `DOCUMENTATION.md`, `README.md` and `CHANGELOG.md`)
- [ ] I’ve bumped the version number (`const VERSION` in `src/Client.php`)

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -384,17 +422,57 @@ try {
...
+The personalisation argument always contains the following required parameters for the letter recipient's address:
+
+- `address_line_1`
+- `address_line_2`
+- `postcode`
+
+Any other placeholder fields included in the letter template also count as required parameters. You need to provide their values in a dictionary with key value pairs. For example:
+
+```php
+personalisation=
```
Comment (leohemsted) [inline DOCUMENTATION.md]: variables should start with $, and all statements (ie - the last line of a code snippet) should end with a semicolon

```php
$personalisation = [
....
];
```

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -629,17 +701,59 @@ An example request would look like:
...
+|`technical-failure`|Email or text message: Notify had a technical failure; you can try to send the message again. <br><br>Letter: Notify had an unexpected error while sending to our printing provider. <br><br>You can omit this argument to ignore this filter.|Yes|Yes||
+|`accepted`|Notify is printing and posting the letter|||Yes|
+|`received`|The provider has received the letter to deliver|||Yes|
+
+#### reference (optional)
+
+A unique identifier you can create if necessary. This reference identifies a single unique notification or a batch of notifications.
+
+```php
+reference='STRING' # optional string - identifies notification(s)
```
Comment (leohemsted) [inline DOCUMENTATION.md]: we have this `reference='STRING'` in the docs four times, each way slightly different syntax/comments - we should probs clean it up to make sure they're all the same

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -1,70 +1,121 @@
...
+    'apiKey' => '{your api key}',
+    'httpClient' => new \Http\Adapter\Guzzle6\Client
+]);
+```
 
-We will show examples here using the Guzzle v6 Adapter.
+To get an API key, [sign in to GOV.UK Notify](https://www.notifications.service.gov.uk/) and go to the __API integration__ page. Refer to the [API keys](#api-keys) section of this documentation for more information.
 
-Setup instructions are also available for [Curl](docs/curl-client-setup.md) and [Guzzle v5](docs/guzzle5-client-setup.md).
+## Guzzle v5
```
Comment (leohemsted) [inline DOCUMENTATION.md]: there's still a lot of shared code between these things. All the text is the same - the only difference is 

`composer require php-http/guzzle5-adapter`
and
```php
'httpClient'    => new \Http\Adapter\Guzzle5\Client(
        new \GuzzleHttp\Client,
        new \Http\Message\MessageFactory\GuzzleMessageFactory
    ),
```

which change between versions. Reckon we can reduce this any more at all?

Comment (jonathanglassman) [reply DOCUMENTATION.md]: User research has suggested that dev users want everything self-contained, so they can just refer to the relevant section, hence the duplication. Happy to chat about.

**Comment by jonathanglassman** [comment]
PHP hasn't passed review yet so could we back out the merge? NET and Node have passed review and are ready for build, hence why i merged them

**Pull request merged**